### ISSUE TITLE
Webhooks

### ISSUE THREAD

**Issue by MDXDave** [zammad/zammad#290]
How about supporting webhooks like [Slack](https://api.slack.com/incoming-webhooks) has it?

**Comment by martini** [comment]
@MDXDave Do you mean for:

a) Webhooks from Zammad -> Slack (e. g. on ticket updates)?
b) Webhooks in general (any time if a object in Zammad has changed, an external URL with the changed object is called)?
c) Webhooks to Zammad -> Called from external to create tickets in Zammad?

**Comment by MDXDave** [comment]
I meant third, but i think the other two options are also very useful.

**Comment by martini** [comment]
a) Will come soon (it's on next release)
b) It's on the roadmap for 2017
c) Any more details? Can you suggest a payload format? What attributes are useful? Should be there a option to update tickets?

**Comment by thorsteneckel** [comment]
Taken from #616 by @martini :

WebHook support like in gitlab or github

<img width="1097" alt="screen shot 2017-01-13 at 13 25 57" src="https://cloud.githubusercontent.com/assets/110226/21929985/e8aa0152-d993-11e6-9113-a7c4636c0032.png">
<img width="1096" alt="screen shot 2017-01-13 at 13 25 00" src="https://cloud.githubusercontent.com/assets/110226/21929986/e8e12484-d993-11e6-9f93-d34e7b557cab.png">

**Comment by dennisoderwald** [comment]
2017 is over - the webhooks in gernal are planned in Q1/2018?

**Comment by plunkettscott** [comment]
Also wondering about webhooks. Would make integration into custom apps way easier.

**Comment by Sebastiandg7** [comment]
Any news about webhooks support?

**Comment by thorsteneckel** [comment]
No. We add the milestone (on the right side of the Issue) when we are working on it.

**Comment by gveltri** [comment]
I just made a custom webhooks app specifically for this purpose.

https://github.com/gveltri/slack-zammad-webhooks

I'd love to work on the feature for Zammad. Does anyone have a working branch?

**Comment by thorsteneckel** [comment]
Hi @gveltri - As far as I know there are currently any branches for this functionality. It would be awesome to receive a pull request for this. You can always contact us here or via a support mail if you need anything or have questions.
There is one thing that is mandatory to get this merged into Zammad: We need a technical solution that brings generic Webhooks. It has to be ensured that we can connect any system in a standardized way. Can't wait to see your first progress  👀🚀

**Comment by MrMontesa** [comment]
I would love to see a webhook that triggers actions in case ticket state changes towards an external app.
here's my use-case:
We have a self developed business app. This one creates new tickets in Zammad via API. Users are working on the ticket and once everything is done, they close the ticket and work continues in business app.
Would be awesome if Zammad could automatically trigger our business app once the ticket state changes to closed.

**Comment by jnshnz** [comment]
Would use webhooks for a better integration with project managment softwares, e.g. with [Taiga](https://github.com/taigaio).

**Comment by gijsrutten** [comment]
Any news/plans regarding Webhooks  in general (any time if a object in Zammad has changed, an external URL with the changed object is called)?

**Comment by MrGeneration** [comment]
We currently have no updates for this, we're currently focusing on knowledgebase and fixing bugs.
We'll keep you updated here when we're ready. :(

**Comment by kwoot** [comment]
Any news on this?

**Comment by MrGeneration** [comment]
Please don't get me wrong, but I'm locking the conversation to contributors only to reduce the noise on this bug tracker.

We will update this issue with more information (like planned release etc) when we're ready to work on this or there's really some news.
